packages/main/src/plugin/index.ts

```diff
@@ -483,6 +483,10 @@ export class PluginSystem {
       undefined,
     );
 
+    apiSender.receive('task-created', () => {
+      statusBarRegistry.setBadged('tasks', true);
+    });
+
```

Comment:
here, it feels to me that we're moving logic to a place where it shouldn't (it would be harder to maintain when some logic code is spread across different classes)

It should be the taskManager that updates its statusBar entry

Reply:
we initialize StatusBarRegistry and TaskManager in index.ts

maybe move initialization of StatusBarRegistry earlier, then in TaskManager we do the 
```
statusBarRegistry.setEntry(
      'tasks',
      false,
      0,
      undefined,
      'Tasks',
      'fa fa-bell',
      true,
      'show-task-manager',
      undefined,
    );
```

this method could return the entry object

then instead of doing somewhere

```ts
 apiSender.receive('task-created', () => {
      statusBarRegistry.setBadged('tasks', true);
    });
```

we do in Taskmanager:

```ts
statusBarTaskEntry.dotted = true;
```

Of course, I would split that enhancement in two:

first, we change order of initialization, return entry object when creating an entry, etc, without introducing any badged/dot/circle

it also means that we could move

```

    commandRegistry.registerCommand('show-task-manager', () => {
      statusBarRegistry.setBadged('tasks', false);
      apiSender.send('toggle-task-manager', '');
    });
```

to the TaskManager instance

so everything managing the icon would be in that class